raiden/transfer/mediated_transfer/mediator.py

```diff
@@ -989,35 +988,51 @@ def handle_init(
     return TransitionResult(iteration.new_state, events)
 
 
-def handle_block(channelidentifiers_to_channels, state, state_change, block_number):
+def handle_block(
+        mediator_state: MediatorTransferState,
+        state_change: Block,
+        channelidentifiers_to_channels: typing.ChannelMap,
+        pseudo_random_generator: random.Random,
+        block_number: typing.BlockNumber,
+):
     """ After Raiden learns about a new block this function must be called to
     handle expiration of the hash time locks.
     Args:
         state (MediatorTransferState): The current state.
     Return:
         TransitionResult: The resulting iteration
     """
-    assert state_change.block_number == block_number
+    balance_proof = mediator_state.transfers_pair[0].payee_transfer.balance_proof
+    channel_identifier = balance_proof.channel_identifier
+    channel_state = channelidentifiers_to_channels.get(channel_identifier)
+
+    secrethash = mediator_state.secrethash
+    locked_lock = channel_state.our_state.secrethashes_to_lockedlocks.get(secrethash)
+    if locked_lock and channel.is_lock_expired(locked_lock, secrethash, block_number):
```

Comment:
you're calling `channelidentifiers_to_channels.get` above, so channel could be none here